docs/ARCHITECHTURE.md

```diff
@@ -0,0 +1,46 @@
+# Batarangle Architecture
+
+The architecture of the Batarangle dev tools extensions can be broken down into 3 parts:
+
+1. **Front-end**: display of block/tree view of the inspected application and other debugging information.
+2. **Middleware**: communication between the inspected application and the dev tools.
+3. **Backend**: lives in the context of the inspected application and notifies the dev tools of any changes to the inspected application through the middleware.
+
+## Frontend
+
+The front-end is an Angular 2.0 application in its own right that lives in the chrome dev tools context. It is based on our version of Flux architecture.
+
+There is a component data store that is notified to the change in the inspected application through the middleware, and exposes an RxJS stream on which the Angular 2.0 component listen to updating the view. In order to be notified to the changes to the inspected application the store subscribes to the BackendActions represented by it's own class.
```

Comment:
**of** the **changes**...on which the Angular 2.0 **components** to **update**....notified **of** the changes